### DISCUSSION TITLE
Input mode in dashboard

### DISCUSSION THREAD

**Discussion by bmichotte** [folke/snacks.nvim#277] [Q&A] [answered]
Hello, 

(thanks folke for all your plugins)

I'm in the process of moving from different plugins to this snacks and I've a strange behavior on the dashboard. When opening nvim, it seems I'm on insert mode, I have to push esc to be able to use the keys. 

Any idea how to debug this ?

Nothing really fancy on the config : https://github.com/bmichotte/dotfiles/blob/develop/config/nvim/lua/plugins/snacks.lua

https://github.com/user-attachments/assets/3d5446ea-9581-46c0-b363-393194b8889e

**Comment by folke** [comment, marked as answer]
You have an autocmd that starts insert mode when a terminal opens...

https://github.com/bmichotte/dotfiles/blob/506471e14af615ebd3982b9a88ae6816238eff4d/config/nvim/lua/commands.lua#L21

**Reply by bmichotte** [reply]
Good catch, thanks @folke

**Comment by maatthc** [comment]
Been facing this for weeks!!!
Thanks both of you!!!
Here what I did to keep my "Insert mode" and not mess up with the dashboard:

```
-- Automatically start insert mode in terminal buffers
-- but not for the terminal of the dashboard
autocmd({ "TermOpen", "BufEnter" }, {
	pattern = { "*" },
	callback = function(ev)
		if  string.find(ev.match, "term") ~= nil then
			vim.cmd(":startinsert")
		end
	end,
	group = group,
})

```
Is there a better way?

Cheers
